addons/mail/static/src/core/common/message.js

```diff
@@ -424,8 +424,15 @@ export class Message extends Component {
     prepareMessageBody(element) {}
 
     enterEditMode() {
+        const body = document.createElement("span");
+        body.innerHTML = this.props.message.body;
+        const mentionedChannelElements = body.querySelectorAll(".o_channel_redirect");
+        const mentionedChannels = Array.from(mentionedChannelElements).map((el) =>
+            this.threadService.getThread(el.dataset.oeModel, el.dataset.oeId)
+        );
```

Comment:
We don't necessarily have local knowledge of these mentioned channels, so this should probably be `Thread.insert()` (or equivalent in 17.0) rather than `Thread.get()`.